Examples/Io/Csv/src/CsvSpacepointsWriter.cpp

```diff
@@ -0,0 +1,58 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2021 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#include "ActsExamples/Io/Csv/CsvSpacepointsWriter.hpp"
+#include "ActsExamples/EventData/SimSpacePoint.hpp"
+#include "Acts/Definitions/Units.hpp"
+
+#include "ActsExamples/Utilities/Paths.hpp"
+
+#include <ios>
+#include <optional>
+#include <stdexcept>
+
+#include <dfe/dfe_io_dsv.hpp>
+
+#include "CsvOutputData.hpp"
+
+ActsExamples::CsvSpacepointsWriter::CsvSpacepointsWriter(
+    const ActsExamples::CsvSpacepointsWriter::Config& cfg,
+    Acts::Logging::Level lvl)
+    : WriterT(cfg.inputSpacepoints, "CsvSpacepointsWriter", lvl), m_cfg(cfg) {
+
+}
+
+ActsExamples::CsvSpacepointsWriter::~CsvSpacepointsWriter() {}
+
+ActsExamples::ProcessCode ActsExamples::CsvSpacepointsWriter::endRun() {
+  // Write the tree
+  return ProcessCode::SUCCESS;
+}
+
+
+ActsExamples::ProcessCode ActsExamples::CsvSpacepointsWriter::writeT(
+    const AlgorithmContext& ctx, const SimSpacePointContainer& spacepoints) {
+
+  // Open per-event file for all components
+  std::string pathSP =
+      perEventFilepath(m_cfg.outputDir, "sp.csv", ctx.eventNumber);
```

Comment:
```suggestion
      perEventFilepath(m_cfg.outputDir, "spacepoints.csv", ctx.eventNumber);
```
would be a bit clearer, I think.